draft-ietf-quic-http.md

```diff
@@ -432,17 +409,76 @@ either by identifying the stream that carries a request or by using a Push ID
 Only a client can send PRIORITY frames.  A server MUST NOT send a PRIORITY
 frame.
 
+## Unidirectional Streams
+
+Unidirectional streams in both directions possess a stream header that indicates
+the stream's purpose.  Some purposes are defined in this document; other
+purposes MAY be defined by extensions to HTTP/QUIC.
+
+~~~~~~~~~~ drawing
+ 0 1 2 3 4 5 6 7
++-+-+-+-+-+-+-+-+
+|Stream Type (8)|
++-+-+-+-+-+-+-+-+
+~~~~~~~~~~
+{: #fig-stream-header title="Unidirectional Stream Header"}
+
+The stream header consists of a single byte, indicating the type of data to
+follow.  The format and structure of that data are defined by the specification
+of the stream type.
+
+If the header indicates a stream type which is not supported by the recipient,
+the stream MUST be aborted (using a QUIC STOP_SENDING frame) with an error of
+type HTTP_UNKNOWN_STREAM_TYPE.  The remainder of the stream SHOULD NOT be read
+by the recipient.
+
+###  Control Streams
 
-## Server Push
+The control stream is indicated by a stream type of `0x43`.  Data on this stream
+consists of HTTP frames, as defined in {{frames}}.
 
-HTTP/QUIC supports server push in a similar manner to {{!RFC7540}}, but uses
+Each side MUST initiate a single control stream at the beginning of the
+connection and send its SETTINGS frame as the first frame on this stream.  Only
+one control stream is permitted; receipt of a second stream which claims to be a
+control stream MUST be treated as a connection error of type
+HTTP_TOO_CONTROLLING.  If the control stream is closed at any point, this MUST
+be treated as a connection error of type HTTP_CLOSED_CRITICAL_STREAM.
+
+Since most connection-level concerns will be managed by QUIC, the primary use of
+the control streams will be for the SETTINGS frame when the connection opens and
+for PRIORITY frames subsequently.
+
+A pair of unidirectional streams is used rather than a single bidirectional
+stream.  This allows either peer to send data as soon they are able.  Depending
+on whether 0-RTT is enabled on the connection, either client or server might be
+able to send stream data first after the cryptographic handshake completes.
+
+### Server Push
+
+HTTP/QUIC server push is similar to what is described in {{!RFC7540}}, but uses
 different mechanisms. During connection establishment, the client enables server
 push by sending a MAX_PUSH_ID frame (see {{frame-max-push-id}}). A server cannot
-use server push until it receives a MAX_PUSH_ID frame.
+use server push until it receives a MAX_PUSH_ID frame.  Only servers can push;
+if a server receives a client-initiated push stream, this MUST be treated as a
+stream error of type HTTP_PUSH_LIMIT_EXCEEDED.
+
+A push stream is indicated by a stream type of `0x50`, followed by the Push ID
```

Comment:
Care to share your rationale for picking these values?

Reply:
If we think a table and 0x00, 0x01, etc. is better, I can do that easily enough.  However, 0x00 in particular seems like a first byte you could get reasonably by accident.  I think non-sequential usage will make it somewhat less likely that bugs/experiments will collide, so I selected these via ASCII values.

Control streams are ASCII "C", push is ASCII "P", and header compression uses "H" and "h".  It has a nice side-effect of viewing the stream data in a UI that renders the bytes into ASCII/UTF-8.

Reply:
Just wondering.  I don't do ASCII conversion in my head, so the numbers looked a little odd.  There is no real reason to pick numbers sequentially.